File: clients/rust/src/models/create_stream_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct CreateStreamIn {
    /// How many bytes in total the stream will retain before dropping data.
    #[serde(rename = "maxByteSize")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub max_byte_size: Option<u64>,

    pub name: String,

    /// How long messages are retained in the stream before being permanently nuked.
    #[serde(rename = "retentionPeriodSeconds")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub retention_period_seconds: Option<u64>,
}

impl CreateStreamIn {
    pub fn new(name: String) -> Self {
        Self {
            max_byte_size: None,
            name,
            retention_period_seconds: None,
        }
    }
}

File: clients/javascript/src/models/createStreamIn.ts
// this file is @generated





export interface CreateStreamIn {
    /** How many bytes in total the stream will retain before dropping data. */
    maxByteSize?: number | null;
name: string;
/** How long messages are retained in the stream before being permanently nuked. */
    retentionPeriodSeconds?: number | null;
}

export const CreateStreamInSerializer = {
    _fromJsonObject(object: any): CreateStreamIn {
        return {
            maxByteSize: object['max_byte_size'],
            name: object['name'],
            retentionPeriodSeconds: object['retention_period_seconds'],
            };
    },

    _toJsonObject(self: CreateStreamIn): any {
        return {
            'max_byte_size': self.maxByteSize,
            'name': self.name,
            'retention_period_seconds': self.retentionPeriodSeconds,
            };
    }
}